src/Shared/NativeMethodsShared.cs

```diff
@@ -627,28 +627,25 @@ internal static bool IsWindows
 #endif
         }
 
-#if MONO
+#if !CLR2COMPATIBILITY
         private static bool? _isOSX;
 #endif
+
         /// <summary>
         /// Gets a flag indicating if we are running under Mac OSX
         /// </summary>
         internal static bool IsOSX
         {
-#if MONO
```

Comment:
Can Mono never CLR2 compatible? If it can be, might need an extra case for this.

Reply:
The compat check is there because that old CLR didn't have nullable structs, the diff makes sense.

Reply:
The CLR2 check is there because that version does not have RuntimeInformation, switching from Mono to that makes sense.